currency/currency

```diff
@@ -136,14 +148,35 @@ checkInternet()
 ## Grabs the exchange rate and does the math for converting the currency
 convertCurrency()
 {
-  exchangeRate=$(httpGet "https://api.exchangeratesapi.io/latest?base=$base" | grep -Eo "$exchangeTo\":[1-9.]*" | grep -Eo "[0-9.]*") > /dev/null
+  trueBase=$base
+  trueTarget=$exchangeTo
+  peggedBase=$( peggedTo "$base" )
+  peggedTarget=$( peggedTo "$exchangeTo" )
+  coef1="1"
+  coef2="1"
+  if [[ "$peggedBase" =~ ^[A-Z]+:[0-9.]+$ ]]; then
+    trueBase=$(echo "$peggedBase" | grep -Eo "^[A-Z]*")
+    coef1=$(echo "$peggedBase" | grep -Eo "[0-9.]*$")
+  fi
+  if [[ "$peggedTarget" =~ ^[A-Z]+:[0-9.]+$ ]]; then
+    trueTarget=$(echo "$peggedTarget" | grep -Eo "^[A-Z]*")
+    coef2=$(echo "$peggedTarget" | grep -Eo "[0-9.]*$")
+  fi
+  if [[ "$trueBase" == "$exchangeTo" || "$base" == "$trueTarget" ]]; then
+    exchangeRate="1"
+  else
+    exchangeRate=$(httpGet "https://api.exchangeratesapi.io/latest?base=$trueBase" | grep -Eo "$trueTarget\":[0-9.]*" | grep -Eo "[0-9.]*") > /dev/null
+  fi
   if ! command -v bc &>/dev/null; then
-    oldRate=$exchangeRate
-    exchangeRate=$(echo $exchangeRate | grep -Eo "^[0-9]*" )
-    amount=$(echo $amount | grep -Eo "^[0-9]*" )
-    exchangeAmount=$(( $exchangeRate * $amount ))
-    exchangeRate=$oldRate
+    exchangeRate=$(echo "$exchangeRate" | grep -Eo "^[0-9]*" )
+    amount=$(echo "$amount" | grep -Eo "^[0-9]*" )
+    coef1=$(echo "$coef1" | grep -Eo "^[0-9]*" )
+    exchangeRate=$(( exchangeRate / coef1 ))
+    exchangeRate=$(( exchangeRate * coef2 ))
+    exchangeAmount=$(( exchangeRate * amount ))
   else
+    exchangeRate=$( echo "scale=8; $exchangeRate / $coef1" | bc )
```

Comment:
Can you explain the purpose of scale here?

Reply:
It's a special variable used inside `bc`, number of decimal places basically. See `$ man bc`, section `VARIABLES`. It's set to zero by default, so quotients end up truncated.

Reply:
Makes sense, thanks for the clarification